mods/doors/init.lua

```diff
@@ -144,9 +144,14 @@ function _doors.door_toggle(pos, node, clicker)
 		local item = clicker:get_wielded_item()
 		local owner = meta:get_string("doors_owner")
 		if item:get_name() == "default:key" then
-			local key_meta = minetest.parse_json(item:get_metadata())
+			local key_meta = item:get_meta()
 			local secret = meta:get_string("key_lock_secret")
-			if secret ~= key_meta.secret then
+
+			if key_meta:get_string("secret") == "" then
+				key_meta:set_string("secret", minetest.parse_json(itemstack:get_metadata()).secret)
```

Comment:
right, this is an error. I should have probably used `itemstack` for consistency in the first place, so maybe we want to change that.